apps/builder/app/builder/features/sidebar-left/panels/pages/search-preview.tsx

```diff
@@ -0,0 +1,155 @@
+import { Flex, Grid } from "@webstudio-is/design-system";
+import { Image, createImageLoader } from "@webstudio-is/image";
+
+/**
+ * Exact google truncation logic is not known, but this is a close approximation
+ */
+export const truncateMetaText = (
+  description: string,
+  maxLength: number = 155
+) => {
+  if (description.length <= maxLength) {
+    return description;
+  }
+
+  const ellipsis = "\u00A0...";
+  let truncated = description.substring(0, maxLength);
+  const lastSpaceIndex = truncated.lastIndexOf(" ");
+
+  // If there's a space to truncate at, use it; otherwise, use the max length
+  truncated =
+    lastSpaceIndex > 0 ? truncated.substring(0, lastSpaceIndex) : truncated;
+
+  return `${truncated}${ellipsis}`;
+};
+
+export const truncateUrl = (pageUrl: string, maxLength = 53) => {
+  if (pageUrl.length <= maxLength) {
+    return pageUrl;
+  }
+
+  const ellipsis = "...";
+  const truncated = pageUrl.substring(0, maxLength);
+
+  return `${truncated}${ellipsis}`;
+};
+
+const formatUrl = (urlString: string) => {
+  const url = new URL(urlString);
+
+  return `${url.origin}${url.pathname.split("/").join(" › ")}`;
+};
+
+type SearchPreviewProps = {
+  /**
+   * The URL of the page to preview in search results
+   */
+  pageUrl: string;
+  title: string;
+  description: string;
+  siteName: string;
+  faviconUrl: string;
+};
+
+/**
+ * ... rotated 90 degrees
+ */
+const VerticalThreePointIcon = () => (
+  <svg
+    height="18"
+    focusable="false"
+    xmlns="http://www.w3.org/2000/svg"
+    viewBox="0 0 24 24"
+  >
+    <path d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"></path>
+  </svg>
+);
```

Comment:
Why not put into icons package?

Reply:
Its not our icon, it will not be reused, its copypasted from google site